director/director.go

```diff
@@ -418,12 +418,32 @@ func redirectToOrigin(ginCtx *gin.Context) {
 	// do the geolocation song and dance if we want to get the closest origin...
 	ipAddr, err := getRealIP(ginCtx)
 	if err != nil {
```

Comment:
This is to be consistent with `redirectToOrigin` behavior, but I agree we should simply fallback to random ordering instead of throwing error if the client IP can't be resolved

Reply:
I've updated this to completely get rid of our `getRealIP` function to use a wrapper around `c.ClientIP()` instead, which is the recommended way to get real IP from a gin request